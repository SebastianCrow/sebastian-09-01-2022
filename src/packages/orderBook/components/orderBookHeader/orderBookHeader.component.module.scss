@import '../../../../styles/variables';
@import '../../../../styles/layout';

.header {
  display: flex;
  align-items: center;
  min-height: 21px;
  padding: $spacing (2 * $spacing);

  .title {
    position: absolute;
  }

  .spreadContainer {
    display: flex;
    align-items: center;
    justify-content: center;
    width: 100%;
  }
}
