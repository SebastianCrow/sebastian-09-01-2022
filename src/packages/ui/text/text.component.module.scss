@import "../../../styles/themes/dark";

.primary {
  color: $color;
}

.secondary {
  color: $color-secondary;
}

.success {
  color: $color-success;
}

.danger {
  color: $color-danger;
}
