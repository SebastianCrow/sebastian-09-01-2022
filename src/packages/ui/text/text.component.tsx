import React, { FunctionComponent } from 'react';
import cns from 'classnames';
import styles from './text.component.module.scss';

export type TextVariant = 'regular' | 'title' | 'code';

export type TextColor = 'primary' | 'secondary' | 'success' | 'danger';

export interface TextProps {
  variant?: TextVariant;
  color?: TextColor;
}

export const Text: FunctionComponent<TextProps> = ({
  variant,
  color = 'primary',
  children,
}) => {
  return (
    <div
      className={cns({
        [styles.title]: variant === 'title',
        [styles.code]: variant === 'code',
        [styles.primary]: color === 'primary',
        [styles.secondary]: color === 'secondary',
        [styles.success]: color === 'success',
        [styles.danger]: color === 'danger',
      })}
    >
      {children}
    </div>
  );
};
