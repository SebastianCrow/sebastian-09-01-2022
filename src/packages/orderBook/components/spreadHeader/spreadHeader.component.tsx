import React, { FunctionComponent } from 'react';
import { Text } from '../../../ui/text/text.component';
import { useSelectSpreadInfo } from '../../hooks/useSelectOrderBookState.hook';

export const SpreadHeader: FunctionComponent = () => {
  const spread = useSelectSpreadInfo();
  // TODO: Placeholder for undefined spread
  if (!spread) {
    return null;
  }
  return (
    <Text color="secondary">{`Spread: ${spread.value} (${spread.percent}%)`}</Text>
  );
};
