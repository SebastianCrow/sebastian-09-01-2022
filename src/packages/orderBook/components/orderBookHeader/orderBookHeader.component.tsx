import React, { FunctionComponent } from 'react';
import { Text } from '../../../ui/text/text.component';
import cns from 'classnames';
import { SpreadHeader } from '../spreadHeader/spreadHeader.component';
import styles from './orderBookHeader.component.module.scss';

export const OrderBookHeader: FunctionComponent = () => {
  return (
    <div className={styles.header}>
      <div className={styles.title}>
        <Text variant="title">Order Book</Text>
      </div>
      <div className={cns(styles.spreadContainer, 'desktop')}>
        <SpreadHeader />
      </div>
    </div>
  );
};
