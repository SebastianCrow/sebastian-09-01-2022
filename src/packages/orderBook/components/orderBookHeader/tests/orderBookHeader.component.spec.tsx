import React from 'react';
import { act, ReactTestRenderer } from 'react-test-renderer';
import { OrderBookHeader } from '../orderBookHeader.component';
import { Text } from '../../../../ui/text/text.component';
import { SpreadHeader } from '../../spreadHeader/spreadHeader.component';
import { createRendererWithOrderBookStore } from '../../../tests/utils/createRendererWithOrderBookStore.util';
import { findByDataTestId } from '../../../../../tests/utils/findByDataTestId.util';

describe('orderBookHeader.component', () => {
  let renderer: ReactTestRenderer;

  test('renders', () => {
    act(() => {
      renderer = createRendererWithOrderBookStore(<OrderBookHeader />);
    });

    expect(renderer.toJSON()).toMatchSnapshot();
    expect(renderer.root.findAllByType(Text)).toHaveLength(1);
    expect(findByDataTestId(renderer.root, 'title')).toBeTruthy();
    expect(renderer.root.findAllByType(SpreadHeader)).toHaveLength(1);
  });

  // TODO: Some kind of visual testing to check media queries (`mobile` and `desktop` classes).
  // E.g. <SpreadHeader /> wrapped in <OrderBookHeader />
});
