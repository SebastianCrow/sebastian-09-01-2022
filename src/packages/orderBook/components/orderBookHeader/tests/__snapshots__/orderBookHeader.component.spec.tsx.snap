// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orderBookHeader.component renders 1`] = `
<div
  className="header"
>
  <div
    className="title"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 title primary css-ahj2mt-MuiTypography-root"
    >
      Order Book
    </p>
  </div>
  <div
    className="spreadContainer desktop"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 secondary css-ahj2mt-MuiTypography-root"
    >
      Spread: 
      17.00
       (
      0.05
      %)
    </p>
  </div>
</div>
`;
